File: lecture_010(RecursionwithArraylists)/l_002_getkeypadcombination.java
import java.util.Scanner;
import java.util.ArrayList;

public class l_002_getkeypadcombination {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        String str = scn.nextLine();
        ArrayList<String> words = getKpc(str);
        System.out.println(words);
    }
    //Global String Array
    static String[] codes = {".;", "abc", "def", "ghi", "jkl", "mno", "pqrs", "tu", "vwx", "yz"};
    public static ArrayList<String> getKpc(String str) {
        if(str.length()==0) {
            ArrayList<String> bres = new ArrayList<>();
            bres.add("");
            return bres;
        }
        char ch = str.charAt(0);  //6
        String ros = str.substring(1);    //78
        ArrayList<String> rres = getKpc(ros);

        ArrayList<String> myans = new ArrayList<>();
        String deep = codes[ch-'0'];
        for(int i=0; i<deep.length(); i++) {
            char ch1 = deep.charAt(i);
            for(String rstr: rres) {
                myans.add(ch1+rstr);
            }
        }
        return myans;
    }
}

File: lecture_010(RecursionwithArraylists)/l_002_getkeypadcombination.cpp
#include<iostream>
#include<vector>
#include<string>
using namespace std;

//Global Array
vector<string> codes = {".;", "abc", "def", "ghi", "jkl", "mno", "pqrs", "tu", "vwx", "yz"};
vector<string> getkpc(string str) {
    if(str.length()==0) {
        vector<string> bres;
        bres.push_back("");
        return bres; 
    }
    char ch = str[0];
    string ros = str.substr(1);
    vector<string> rres = getkpc(ros);

    vector<string> myans;
    string deep = codes[ch-'0'];
    for(int i=0; i<deep.length(); i++) {
        char ch1 = deep[i];
        for(string rstr: rres) {
            myans.push_back(ch1+rstr);
        }
    }
    return myans;
}

int main() {
    string str;
    //cin>>str;
    getline(cin,str);
    vector<string> words = getkpc(str);
    cout<<"[";
    for(int i=0; i<words.size()-1; i++) {
        cout<<words[i]<<", ";
    }
    cout<<words[words.size()-1]<<"]";
    
    return 0;
}